app/assets/javascripts/delete_ideas.js

```diff
@@ -0,0 +1,11 @@
+var deleteIdeas = function(requestService) {
+  $('#ideas').on('click', '.delete', function(){
+    var idea = this.closest('.idea')
+    deleteIdeaInDatabase(idea, requestService).then($(idea).remove())
```

Comment:
The `$(idea).remove()` in  `.then($(idea).remove())` removes the idea immediately. You probably didn't notice this because it's unlikely to fail. But, those parentheses call it immediately. You should wrap it in an anonymous function to prevent it from immediately executing.

```js
deleteIdeaInDatabase(idea, requestService).then(function () { $(idea).remove() });
```